### DISCUSSION TITLE
`git log -L` from `magit-file-dispatch`

### DISCUSSION THREAD

**Discussion by astoff** [magit/magit#4479] [Support] [answered]
Suppose I want to see which commits changed the current line of a file. Calling `M-x magit-file-dispatch RET L -L` isn't a very streamlined process:

1) I get asked which file I want to trace changes from, when it's clear that I will choose the current file
2) I get asked which lines I'm interested in, when it's almost certain I'm interested in the current line, or maybe the range of lines of the active region
3) I have to manually clear the "Limit to files"  (`--`) switch, because git expects no pathspec limiters to be provided together with the `-L` option.

Instead, I would expect that `M-x magit-file-dispatch RET L -L RET` directly shows all commits that changed the current line of the current buffer's file. Or perhaps `M-x magit-file-dispatch RET L -L RET RET`, where the next to last `RET` selects a default line range.

**Comment by tarsius** [comment, marked as answer]
1. Don't us `M-x` to invoke the dispatch command. The default binding is `C-x M-g` but that's not very good either. I recommend `C-c f` but packages are not allowed to use the `C-c <letter>` space so you have to add this binding yourself.
2. In the dispatch menu do not invoke the "do everything explicitly" log command (`L`), but the "dwim" log command (`l`). That shows the log for the current file at least.
3. Use `t`, that shows the log for the current definition.
4. Mark the region (e.g. just the current line) and use `l`. Tadaa, log for that region/line.

**Reply by astoff** [reply]
I see. This is pretty intuitive, I guess I could have guessed it.

BTW, `M-x magit-file-dispatch RET` is not my usual keybinding to that command :-).
